V12

sandervspl opened the pull request:
- Ik heb bij deze update ook de manier van assets importen veranderd. Next doet een aantal dingen met static assets en dat gebeurde niet met de manier waarop ik Webpack had geconfigureerd. Om dit nu bij elke update steeds aan te gaan passen is niet te doen, dus ik heb onze Webpack regels voor images (behalve SVG) verwijderd. Om nu een static image te gebruiken voeg je ze toe aan `public/images` en import je ze met `/images/my-image.png` in bijvoorbeeld de src van de `<Image />` Next component of in CSS bij bijvoorbeeld `url()`. Ik heb deze change nu ook vermeld in de README.
- Ik heb `next-offline` vervangen met `next-pwa` omdat de eerste niet meer actief wordt geupdated en de nieuwe wel.
- De nieuwe Rust compiler kunnen wij helaas nog niet gebruiken. Next gebruikt SWC om Babel te vervangen als compiler, maar helaas heeft deze nog niet volledige support voor Styled-Components. Ik heb geen issues gezien, maar om het safe te spelen heb ik SWC niet enabled en zal je nog geen speed improvements zien.

sandervspl committed:
Update to Next v12 & update to import file as url

sandervspl committed:
Add Sharp, replace next-offline with next-pwa

sandervspl committed:
Update README.md

rnnyrk commented on tsconfig.json:
```diff
@@ -17,30 +17,69 @@
+      ],
+      "vectors/*": [
+        "src/static/vectors/*"
+      ]
```
Maybe nice to put all of these on one line? I know eslint copy paste rules do mess this up tho

```json
"types": ["src/types/index"
"vectors/*": ["src/static/vectors/*"]
```

sandervspl replied:
Fixed. Next messed up the file when I first launched the server with v12.

rnnyrk commented on package.json:
```diff
@@ -20,53 +20,55 @@
-    "eslint": "7.29.0",
-    "eslint-config-next": "11.0.1",
+    "copy-webpack-plugin": "9.0.1",
+    "esbuild": "0.13.12",
```
Just curious; Why is this project both `esbuild` and `webpack`?

sandervspl replied:
Esbuild was added into the v11 release of react-prime and is used to compile a couple of things to JS
- The next.config.ts file. Next does not support a .ts config file out of the box.
- The custom Next server

Both these things would eat up seconds of startup time when building with any other bundler so esbuild is included to compile both these things in milliseconds.

Webpack is used for the Next config.

Commit:
Merge branch 'master' into v12

sandervspl committed:
Restore babel config

SWC compiler seems to lack styled-components support. Although I have not seen issues with it, it is recommended to keep using Babel if you use styled-components until a later update.

sandervspl committed:
Update tsconfig.json

sandervspl committed:
Update packages

sandervspl committed:
Disable 'explicit-module-boundary-types' rule
